apollo-federation/src/merger/merger.rs

```diff
@@ -199,13 +239,677 @@ impl Merger {
     fn post_merge_validations(&mut self) {
         todo!("Implement post-merge validations")
     }
+
+    /// Merge enum type from multiple subgraphs
+    pub(crate) fn merge_enum(
+        &mut self,
+        sources: Sources<&EnumType>,
+        dest: &mut EnumType,
+    ) -> Result<(), FederationError> {
+        let usage = self.enum_usages.get(&dest.name.to_string()).cloned().unwrap_or_else(|| {
```

Comment:
You don't need to create a new string here:
```suggestion
        let usage = self.enum_usages.get(dest.name.as_str()).cloned().unwrap_or_else(|| {
```